ios/engine/KMEI/KeymanEngine/resources/Keyman.bundle/Contents/Resources/keyboard.html

```diff
@@ -272,8 +272,13 @@
                 if(undefined == text) text = '';
                 var ta = document.getElementById('ta');
                 var kmw = window['keyman'];
+
+                let resetContext = ta.value != text
                 ta.value = text;
                 kmw['setActiveElement'](ta);
+                if(resetContext) {
```

Comment:
Won't this also need to go in KMEA keyboard.html (`updateKMText()`)?

Reply:
It... might.  It also might just be me being extra cautious; not sure which.  Testing should clarify this.

I made the main breakthrough late in the workday, so any personal tests on my end regarding this will need to wait 'til tomorrow.

Reply:
And... it does.  Added in the latest commit.